### DISCUSSION TITLE
Use of nonlinear slider as crossfader pot

### DISCUSSION THREAD

**Discussion by seedlord** [tttapa/Control-Surface#658] [Q&A] [answered]
Hello,

as the title says, i wonder how a nonlinear slider will behave as a pot.
i have a stock 45mm fader from reloop spin, that i plan to make into a usb fader. 
its main purpose should be fast cutting in dvs software.
measuring the resistance shows 4.4 Ohm at smallest point. 
~8.55 kOhm at 25%. 
~18 kOhm at 50%. 
~23 kOhm at 75%.
23.7 kOhm at 100%. 
there is a peak to 24kOhm at ~95%. (that seems to be the cut point for the reloop spin)

in my daw i can program a custom crossfader curve, so using the 24kOhm peak should be possible.
but will this lead into any problems in arduino or in the code or in the output as the curve is non linear?
do i need another fader or can i use the reloop stock fader?

thanks in advance

**Comment by tttapa** [comment, marked as answer]
You can use the mapping function, see the [CCPotentiometer-Map.ino](https://tttapa.github.io/Control-Surface-doc/Doxygen/d3/d26/CCPotentiometer-Map_8ino-example.html) example.

Logarithmic potentiometers are usually closer to piecewise linear:
![Log pot curve](https://user-images.githubusercontent.com/21296729/41199887-461e16d0-6c9a-11e8-8f3d-168eea9b9012.png)

You could use a mapping function like the one used here: https://github.com/tttapa/MIDI_controller/issues/109#issuecomment-698823253

Do note that applying such a mapping will not be perfect, and it will amplify noise, so it might be better to get a true linear fader if possible.

**Reply by seedlord** [reply]
Thanks for you answer sir.

I built the fader into a arduino nano clone and i use hairless/loopmidi to make usbmidi from serial. it works great.
tested quickly in daw and it seems to behave linear. 
i will test more and report back.
